src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/PlatformDetection.cs

```diff
@@ -168,21 +168,28 @@ private static bool GetSsl3Support()
             {
                 string clientKey = @"HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\SSL 3.0\Client";
                 string serverKey = @"HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\SSL 3.0\Server";
-                bool enabled = true;
 
-                // This may change in future but for now, missing key means protocol is enabled.
+                object client, server;
                 try
                 {
-                    if ((int)Registry.GetValue(clientKey, "Enabled", 1) == 0 || (int)Registry.GetValue(serverKey, "Enabled", 1) == 0)
-                    {
-                        enabled = false;
-                    }
+                    client = Registry.GetValue(clientKey, "Enabled", null);
+                    server = Registry.GetValue(serverKey, "Enabled", null);
                 }
-                catch (Exception e) when (e is SecurityException || e is InvalidCastException || e is NullReferenceException)
+                catch (SecurityException)
+                {
+                    // Insufficient permission, assume that we don't have SSL3 (since we aren't exactly sure)
+                    return false;
+                }
+
+                if (client is int c && server is int s)
                 {
+                    return c == 1 && s == 1;
                 }
 
-                return enabled;
+                // Missing key. If we're pre-20H1 then assume SSL3 is enabled.
+                // Otherwise, disabled. (See Comments on dotnet/runtime#1166)
+                // Alternatively the returned values must have been some other types.
+                return !IsWindows10Version2004OrGreater;
```

Comment:
would it make sense to fall back to previous logic? It seems like this change is needed because default changed, but we could still respect registry on older versions.

Reply:
The previous logic if I understood correctly is:
* if the key exists & is set to 0, disable it
* If the key does not exist / is of a invalid type or we don't have access, assume default

The Current logic is:
* if the key exists & is set to 0, disable it
* If the key does not exist / is of a invalid type:
  * If we're pre-20H1, default to enabled
  * If we're post-20H1, default to disabled
* If we don't have access to the key, assume it's disabled (because theoretically pre-20H1 builds still can have it disabled by setting the key, and having no access means we don't know if it is enabled or disabled; I wouldn't want to cause test failures in those cases by defaulting to enabled)

I'm not seeing how we're not respecting registry on older versions?